pandas/core/base.py

```diff
@@ -894,12 +906,36 @@ def to_numpy(self):
         >>> ser = pd.Series(pd.Categorical(['a', 'b', 'a']))
         >>> ser.to_numpy()
         array(['a', 'b', 'a'], dtype=object)
+
+        Specify the `dtype` to control how datetime-aware data is represented.
+        Use ``dtype=object`` to return an ndarray of pandas :class:`Timestamp`
+        objects, each with the correct ``tz``.
+
+        >>> ser = pd.Series(pd.date_range('2000', periods=2, tz="CET"))
+        >>> ser.to_numpy(dtype=object)
```

Comment:
Note that the *default* behavior right now is the same as `dtype="datetime64[ns]"`, i.e. the timezone info is lost. I don't think that's what we want, but I'm waiting on #24024 to be done before making that change.

I also have a branch (https://github.com/TomAugspurger/pandas/pull/new/dt-array-3) that's deprecating the behavior for Series.__array__ and Index.__array__ returning datetime64[ns] for tz-aware values. That's currently blocked by #24024.